### DISCUSSION TITLE
`crc16_reflect()` showing unexpected results

### DISCUSSION THREAD

**Discussion by tomschmitz98** [zephyrproject-rtos/zephyr#78135] [Q&A] [answered]
Hello, I am migrating a project to Zephyr and I noticed there was a CRC module in Zephyr. Since CRC16 is being used in the project that I am working on, I decided to test the module and found that `crc16_reflect()` was not outputting the expected value. I was wondering if I am misusing the library or if this was a genuine bug that I discovered.

example code:

```c
#define POLYNOMIAL 0xD175

uint8_t src[] = "Hello, World.";
size_t len = sizeof(src) - 1;

uint16_t crc = crc16_reflect(POLYNOMIAL, 0, src, len);

// Expected value of crc: 0xF97E
// Received value of crc: 0x5F50
```

I checked the result with several web CRC calculators such as [sunshine's CRC calculator](http://www.sunshine2k.de/coding/javascript/crc/crc_js.html)

Parameters:
- Input and output reflect
- Polynomial: 0xD175
- Initial value: 0
- Final XOR value: 0

**Comment by kartben** [comment, marked as answer]
As per the documentation of the function, the polynomial should be reflected.

> The polynomial to use omitting the leading x^16 coefficient. Important: please reflect the poly. 

Using 0xae8b (0xd175 reflected) in the web calculator matches the output you're seeing :)

**Reply by tomschmitz98** [reply]
Thanks you, that works! I missed that in the docs :/
